systemtest/processor_test.go

```diff
@@ -141,44 +142,62 @@ func TestHotStandby(t *testing.T) {
 	// check the table-values
 	val1, _ := tableStorage1.Get("key1")
 	val2, _ := tableStorage2.Get("key2")
-	test.AssertEqual(t, string(val1), "message1")
-	test.AssertEqual(t, string(val2), "message2")
+	require.Equal(t, string(val1), "message1")
+	require.Equal(t, string(val2), "message2")
 
 	// check the join-values
 	joinval1, _ := joinStorage1.Get("key1")
 	joinval2, _ := joinStorage2.Get("key2")
-	test.AssertEqual(t, string(joinval1), "joinval1")
-	test.AssertEqual(t, string(joinval2), "joinval2")
+	require.Equal(t, string(joinval1), "joinval1")
+	require.Equal(t, string(joinval2), "joinval2")
 
 	// stop everything and wait until it's shut down
 	cancel()
-	test.AssertNil(t, errg.Wait().ErrorOrNil())
+	require.NoError(t, errg.Wait().ErrorOrNil())
 }
 
 // Tests the processor option WithRecoverAhead. This requires a (local) running kafka cluster.
 // Two processors are initialized, but they share an input topic which has only one partition. This
 // Test makes sure that still both processors recover the views/tables
 func TestRecoverAhead(t *testing.T) {
 	brokers := initSystemTest(t)
-
 	var (
-		group       goka.Group = "goka-systemtest-recoverahead"
-		inputStream string     = string(group) + "-input"
-		table                  = string(goka.GroupTable(group))
-		joinTable   goka.Table = "goka-systemtest-recoverahead-join"
+		group       = goka.Group(fmt.Sprintf("goka-systemtest-recoverahead-%d", time.Now().Unix()))
+		inputStream = fmt.Sprintf("%s-input", group)
+		table       = string(goka.GroupTable(group))
+		joinTable   = goka.Table(fmt.Sprintf("%s-join", group))
 	)
 
 	tmc := goka.NewTopicManagerConfig()
 	tmc.Table.Replication = 1
 	tmc.Stream.Replication = 1
 	cfg := goka.DefaultConfig()
+	cfg.Consumer.Offsets.Initial = sarama.OffsetOldest
+	goka.ReplaceGlobalConfig(cfg)
+
 	tm, err := goka.TopicManagerBuilderWithConfig(cfg, tmc)(brokers)
-	test.AssertNil(t, err)
+	require.NoError(t, err)
 
 	err = tm.EnsureStreamExists(inputStream, 1)
-	test.AssertNil(t, err)
+	require.NoError(t, err)
```

Comment:
this test was flaky altogether, because it created the wrong topic, which ended up with two topics having two different number of partitions, as the examples-cluster was configured with "auto-create:true". This is fixed now and yields two new issues:
* fix all examples that depend on "auto-create:true" (#379)
* detect partition mismatches (#378)